pkg/operator/controllers/subnets/subnets_controller.go

```diff
@@ -101,7 +102,34 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		subnets:        subnet.NewManager(&azEnv, resource.SubscriptionID, authorizer),
 	}
 
-	return reconcile.Result{}, manager.reconcileSubnets(ctx)
+	return reconcile.Result{}, manager.reconcileSubnets(ctx, instance)
+}
+
+func (r *reconcileManager) reconcileSubnets(ctx context.Context, instance *arov1alpha1.Cluster) error {
+	subnets, err := r.kubeSubnets.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	// This calls potentially calls update 2x for same loop, but this is price
+	// to pay to keep logic split, separate and simple
+	for _, s := range subnets {
```

Comment:
Do we want to error out completely in this loop? If we fail on a subnet then we should still try and do the other ones, rather than potentially having #1 succeed, #2 fail, and then #3 never be touched, even though it would succeed. I think instead we should just save the error and return it in a list once we've tried them all.

Reply:
Changed the logic to continue on error and return a combined error at the end if any occurred.